quinn/src/platform/unix.rs

```diff
@@ -198,9 +198,19 @@ fn send(io: &mio::net::UdpSocket, transmits: &[Transmit]) -> io::Result<usize> {
     let mut msgs: [libc::mmsghdr; BATCH_SIZE] = unsafe { mem::zeroed() };
     let mut iovecs: [libc::iovec; BATCH_SIZE] = unsafe { mem::zeroed() };
     let mut cmsgs = [cmsg::Aligned([0u8; CMSG_LEN]); BATCH_SIZE];
+    let mut addrs: [MaybeUninit<socket2::SockAddr>; BATCH_SIZE] =
+        unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
I agree with @faern that this looks pretty surprising. Can you add a comment outlining your take on why this is okay, maybe referring to the unstable features that might help here in the future?